packages/expo-apple-authentication/src/AppleAuthentication.types.ts

```diff
@@ -0,0 +1,248 @@
+ * user whose credentials you'd like to refresh.
+ *
+ * @see [Apple
+ * Documentation](https://developer.apple.com/documentation/authenticationservices/asauthorizationopenidrequest)
+ * for more details.
+ */
+export type AppleAuthenticationRefreshOptions = {
+  user: string;
+
+  /**
+   * The scope of personal information to which your app is requesting access.
+   * The user can choose to deny your app access to any scope at the time of
+   * refreshing.
+   * @defaults `[]` (no scopes).
+   */
+  requestedScopes?: AppleAuthenticationScope[];
+
+  /**
+   * Data that’s returned to you unmodified in the corresponding credential
+   * after a successful authentication. Used to verify that the response was
+   * from the request you made. Can be used to avoid replay attacks.
+   */
+  state?: string;
+}
+
+/**
+ * The options you can supply when making a call to
+ * `AppleAuthentication.logout()`. You must include the ID string of the user to
+ * sign out.
+ *
+ * @see [Apple
+ * Documentation](https://developer.apple.com/documentation/authenticationservices/asauthorizationopenidrequest)
+ * for more details.
+ */
+export type AppleAuthenticationLogoutOptions = {
+  user: string;
+
+  /**
+   * Data that’s returned to you unmodified in the corresponding credential
+   * after a successful authentication. Used to verify that the response was
+   * from the request you made. Can be used to avoid replay attacks.
+   */
+  state?: string;
+}
+
+/**
+ * The user credentials returned from a successful call to
+ * `AppleAuthentication.loginAsync()`, `AppleAuthentication.refreshAsync()`, or
+ * `AppleAuthentication.logoutAsync()`.
+ *
+ * @see [Apple
+ * Documentation](https://developer.apple.com/documentation/authenticationservices/asauthorizationappleidcredential)
+ * for more details.
+ */
+export type AppleAuthenticationCredential = {
+  /**
+   * A value indicating the status type of the requested credential. `SUCCESS`
+   * if the credential was retrieved successfully, or `CANCEL` if the user
+   * canceled the operation before signing in.
+   */
+  type: AppleAuthenticationStatus;
+
+  /**
+   * An identifier associated with the authenticated user. You can use this to
+   * check if the user is still authenticated later. This is stable and can be
+   * shared across apps released under the same development team. The same user
+   * will have a different identifier for apps released by other developers.
+   */
+  user?: string;
+
+  /**
+   * An arbitrary string that your app provided as `state` in the request that
+   * generated the credential. Used to verify that the response was from the
+   * request you made. Can be used to avoid replay attacks.
+   */
+  state?: string;
+
+  /**
+   * The user’s name. May be null if you didn't request the `FULL_NAME` scope or
+   * if the user denied access. May also be null if this is not the first time
+   * the user has signed into your app.
+   */
+  fullName?: AppleAuthenticationFullName;
```

Comment:
For these nullable fields, we should use a union type with `null` instead of an optional property (which does not include `null` as a valid value). We do need to verify that the platform layer always defines the property, though.

```suggestion
  fullName: AppleAuthenticationFullName | null;
```

Reply:
Actually as we changed our mind in terms of cancellation API, then we can say that `fullName`, `user`, `identityToken` and maybe some other fields can't be null/undefined 🎉 I'm gonna change them now.